### PR TITLE
feat: include claude 3 adapter

### PR DESCRIPTION
**Pull request by albertsyh**
- [x] Separated text and messages completions API for claude
- [x] Older claude models are accessed via `getClaudeLegacyAdapter` [here](https://github.com/hrishioa/wishful-search/blob/2c7514da1b8bb5221171c707d6d08a455ba58a52/src/llm-adapters.ts#L168)

### CODE REVIEW

**Review by hrishioa — APPROVED** [review]

File: src/llm-adapters.ts [hunk 1]
```diff
@@ -252,6 +253,89 @@ function getClaudeAdapter(
...
+    callLLM: LLMCallFunc;
+  } = {
+    llmConfig: DEFAULT_CLAUDE_LLM_CONFIG,
+    callLLM: async function callLLM(
+      messages: LLMCompatibleMessage[],
+      queryPrefix?: string,
+    ) {
+      try {
+        const formattedMessages = messages.filter(
+          (message) => message.role !== 'system',
```
Comment (hrishioa) [inline src/llm-adapters.ts:278]: you need to remove any trailing spaces with a `trimEnd` from the last message if it's from the assistant - something weird Claude has now

**Review by hrishioa — APPROVED** [review]
One comment otherwise good to go!

File: src/llm-adapters.ts (hunk 1 above)
Comment (as-greywing) [reply src/llm-adapters.ts:278]: I'll trim it from all assistant messages, irrespective of whether it's the last. that shouldn't be a problem right?

Comment (hrishioa) [reply src/llm-adapters.ts]: Yup that works too

**Pull request merged**